src/FluentValidation/TestHelper/ValidatorTestExtensions.cs

```diff
@@ -66,6 +79,19 @@ public static class ValidationTestExtension {
 			testValidationResult.ShouldNotHaveValidationErrorFor(expression);
 		}
 
+		public static void ShouldNotHaveValidationErrorFor<T>(this IValidator<T> validator, Expression<Func<T, T>> expression, string ruleSet)

```

Comment:
Same as above, let's remove this overload as it doesn't accept the model instance which leads to confusing behaviour.